CHANGELOG.md

```diff
@@ -27,6 +27,7 @@
 
 * Fixed an issue where changes made to `NavigationViewController.showsReportFeedback`, `NavigationViewController.showsSpeedLimits`, `NavigationViewController.detailedFeedbackEnabled`, `NavigationViewController.floatingButtonsPosition` and `NavigationViewController.floatingButtons` before `NavigationViewController` presentation were not saved. ([#3718](https://github.com/mapbox/mapbox-navigation-ios/pull/3718))
 * Fixed an issue where `SpeechSynthesizing.managesAudioSession` was ignored by `RouteVoiceController`. ([#3572](https://github.com/mapbox/mapbox-navigation-ios/pull/3572))
+* Location tracking and routing resources are now freed if no `RouteController` or `PassiveLocationManager` instance is actively being used. ([#3724](https://github.com/mapbox/mapbox-navigation-ios/pull/3724))
```

Comment:
- [ ] We need to expand this changelog entry saying the `startRecordingHistory`, `pushHistoryEvent`, doesn't work if there is no active passive or active navigation. 

- [ ] Documentation for these methods to be updated.

- [ ] And given that these methods are now depend on navigator to exist, it makes sense to consider to deprecate them and make them work on an instance.

Reply:
https://github.com/mapbox/mapbox-navigation-ios/issues/3753